torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,448 @@
+    def delta(self, other, inclusive: bool = False, subtract_baselines: bool = True):
+        """Diff two sets of counts.
+
+        One common reason to collect instruction counts is to determine the
+        the effect that a particular change will have on the number of instructions
+        needed to perform some unit of work. If a change increases that number, the
+        next logical question is "why". This generally involves looking at what part
+        if the code increased in instruction count. This function automates that
+        process so that one can easily diff counts on both an inclusive and
+        exclusive basis. The `subtract_baselines` argument allows one to disable
+        baseline correction, though in most cases it shouldn't matter as the
+        baselines are expected to more or less cancel out.
+        """
+        if subtract_baselines:
+            first = self.stats(inclusive=inclusive)
+            second = other.stats(inclusive=inclusive)
+        else:
+            if inclusive:
+                first, second = self.stmt_inclusive_stats, other.stmt_inclusive_stats
+            else:
+                first, second = self.stmt_exclusive_stats, other.stmt_exclusive_stats
+        return self._diff(first, second)
+
+    # FIXME: Once 3.7 is the minimum version, type annotate output
+    def as_standardized(self):
+        """Strip some prefixes from function strings.
+
+        When comparing two different sets of instruction counts, on stumbling
+        block can be path prefixes. Callgrind includes the full filepath
+        when reporting a function (as it should). However, this can cause
+        issues when diffing profiles. If a key component such as Python
+        or PyTorch was built in separate locations in the two profiles, which
+        can result in something resembling:
+            23234231 /tmp/first_build_dir/thing.c:foo(...)
+             9823794 /tmp/first_build_dir/thing.c:bar(...)
+              ...
+               53453 .../aten/src/Aten/...:function_that_actually_changed(...)
+              ...
+             -9823794 /tmp/second_build_dir/thing.c:bar(...)
+            -23234231 /tmp/second_build_dir/thing.c:foo(...)
+
+        Stripping prefixes can ameliorate this issue by regularizing the
+        strings and causing better cancelation of equivilent call sites
+        when diffing.
+        """
+        def strip_prefix(stats: Tuple[Tuple[int, str], ...]) -> Tuple[Tuple[int, str], ...]:
+            counts: DefaultDict[str, int] = collections.defaultdict(int)
+
+            # "Python" and "Objects" come from CPython.
+            prefix_truncations = ("build/aten/", "Python/", "Objects/")
+            for c, fn in stats:
+                fn = re.sub(r"^.+build/\.\./", "build/../", fn)
+                for new_prefix in prefix_truncations:
+                    fn = re.sub(r"^.+" + new_prefix, new_prefix, fn)
```

Comment:
NB: this will do stripping even when the truncation occurs in the middle of a name. So for example, `foo/MyPython/blah` would get truncated to `Python/blah` because of how the regex is written

Reply:
Also, minor hygiene thing: use `re.escape` here

Reply:
Good points.